src/motion/space_groups.F

```diff
@@ -0,0 +1,1229 @@
+     WRITE (UNIT=fmtstr1(22:23), FMT="(I2)") n + 7
+     fmtstr2 = "(T3,A,T5,3(1X,F  .  ))"
+     WRITE (UNIT=fmtstr2(16:17), FMT="(I2)") n + 7
+     WRITE (UNIT=fmtstr2(19:20), FMT="(I2)") n
+     WRITE (UNIT=spgr%punit, FMT=fmtstr1) &
+           "SPGR| SYMMETRIZED STRESS TENSOR [GPa]", "X", "Y", "Z"
+     WRITE (UNIT=spgr%punit, FMT=fmtstr2) "X", stress_tensor(1, 1:3)
+     WRITE (UNIT=spgr%punit, FMT=fmtstr2) "Y", stress_tensor(2, 1:3)
+     WRITE (UNIT=spgr%punit, FMT=fmtstr2) "Z", stress_tensor(3, 1:3)
+
+     CALL timestop(handle)
+
+   END SUBROUTINE spgr_write_stress_tensor
+
+! **************************************************************************************************
+!> \brief routine prints cell details. Only for debugging purpose.
+!> \param cell ...
+!> \param iter ...
+!> \param io output unit
+!> \par History
+!>      01.2020 created [pcazade]
+!> \author Pierre-André Cazade (first version)
+! **************************************************************************************************
+   SUBROUTINE spgr_print_cell(cell,iter,io)
+      IMPLICIT NONE
+      
+      INTEGER                                            :: iter,io
+      TYPE(cell_type)                                    :: cell
+
+      WRITE (io, *) "SPGR| ITERATION: ",iter
+      WRITE (io, *) "SPGR| CELL_REF| A: ", cell%hmat(:,1)*0.52917721067
+      WRITE (io, *) "SPGR| CELL_REF| B: ", cell%hmat(:,2)*0.52917721067
+      WRITE (io, *) "SPGR| CELL_REF| C: ", cell%hmat(:,3)*0.52917721067
+
+   END SUBROUTINE spgr_print_cell
+
+! **************************************************************************************************
+!> \brief routine prints coordinates or forces. Only for debugging purpose.
+!> \param x ...
+!> \param iter ...
+!> \param io output unit
+!> \par History
+!>      01.2020 created [pcazade]
+!> \author Pierre-André Cazade (first version)
+! **************************************************************************************************
+   SUBROUTINE spgr_print_xg(x,iter,io)
+      IMPLICIT NONE
+
+      INTEGER                                            :: iter,nparticle,i,ia,io
+      REAL(KIND=dp), DIMENSION(:)                        :: x
+
+      nparticle=size(x)/3
+      write(io,*) nparticle
+      write(io,*) "Iteration: ",iter
+      do i = 1,nparticle
+         ia = (i-1)*3
+         write(io,*) x(ia+1)*0.52917721067,x(ia+2)*0.52917721067,x(ia+3)*0.52917721067
+      end do
+
+   END SUBROUTINE spgr_print_xg
+
+! **************************************************************************************************
+!> \brief routine prints hessian matrix. Only for debugging purpose.
+!> \param hess ...
+!> \param iter ...
+!> \param io output unit
+!> \par History
+!>      01.2020 created [pcazade]
+!> \author Pierre-André Cazade (first version)
+! **************************************************************************************************
+   SUBROUTINE spgr_print_hess(hess,iter,io)
```

Comment:
please remove if not used, or hide them (with proper format strings) behind a print key if you think they could be useful for others